src/vegur_client.erl

```diff
@@ -690,7 +690,8 @@ stream_header(Client=#client{state=State, buffer=Buffer,
                             end
                     end;
                 <<"connection">> ->
-                    Values = header_list_values(Value),
```

Comment:
Would propose to get rid of `header_list_values` as it is only used in one other place. Also wondering if we can test this?